inc/nu/impl/proclet.ipp

```diff
@@ -73,6 +79,26 @@ retry:
     Migrator::migrate_thread_and_ret_val<void>(
         std::move(return_buf), to_proclet_id(caller_header), nullptr, nullptr);
   }
+  else{
+    // metric logging
+    if (caller_header){
+      ProcletSlabGuard slab_guard(&caller_header->slab);
+      NodeIP target_ip = get_runtime()->rpc_client_mgr()->get_ip_by_proclet_id(id);
+
+      caller_header->spin_lock.lock();
```

Comment:
This will limit the invocation tput of the caller proclet to roughly 1MOPS

Reply:
> 

Is this because of the call to get_ip_by_proclet_id or the locking? The locking can theoretically be removed by replacing the map with an array that can be unsafely modified.

Reply:
I mean the locking. Yeah making it lockless would be better.